src/main/scala/viper/silver/ast/pretty/PrettyPrinter.scala

```diff
@@ -624,6 +624,7 @@ object FastPrettyPrinter extends FastPrettyPrinterBase with BracketPrettyPrinter
       case dt@DomainType(domainName, typVarsMap) =>
         val typArgs = dt.typeParameters map (t => show(typVarsMap.getOrElse(t, t)))
         text(domainName) <> (if (typArgs.isEmpty) nil else brackets(ssep(typArgs, char (',') <> space)))
+      case SMTType(boogieName, _) => boogieName
```

Comment:
Is it really the Boogie name we want to use? Seems odd, since the extension is about SMT types.

Reply:
I figured the Boogie name would be less confusing because SMT type names can look a bit weird. `float24e8` just looks more like a type to me than `(_ FloatingPoint  8  24)`. But I don't really have a strong opinion on that.

Reply:
I agree, the SMT type name is really no eye candy. Does the parser also expect the Boogie name format?

Maybe we could only use the Boogie type names, and backends (Silicon) would have to map them to SMT, if necessary, potentially with the help of some Silver function.

Reply:
This is currently only in the AST for frontends to use, there is no support for this on the source level, the parser doesn't know about this.
Yeah sure, we could do that. Then we would lose the option of using types that exists in Z3 but not in Boogie, though, but I guess that would be okay.

Reply:
If we wanted to merge more or less as-is, we could probably describe the feature as (very) experimental, and see what happens. Or just not document it at all :-)